andela/ah-backend-odin
# Ft registration email confirmation 161340174

#### What does this PR do?
Ensures that user receives verification email.


#### Description of Task to be completed?
When a user signs up successfully they are sent a confirmation email after following the steps below:

`POST  api/users/`

`GET api/users/verify/<uidb64>/<token>/`


#### How should this be manually tested?
After cloning the repo to your machine, cd into it and run this snippet in the terminal 
`python manage.py runserver`

Open your Postman or Postico and feed it with the following url
`http://127.0.0.1:8000/api/users/`

The `BODY` of your `POST` request should contain the following content
`{
  "user":{
    "username": "user",
    "email": "user@user.user",
    "password": "userpassword"
  }
}`

After having submitted valid data a user account is created but isn't activated yet, this can be confirmed by running the following query through the database and observing that the column `is-active` will be set to `false`.
`SELECT * FROM public.authentication_user
ORDER BY id DESC `

When the user verifies their account using the confirmation email the status of `is_active` will change to `true` and this can be observed by rerunning the query above.

The user will now be able to log in.


#### Any background context you want to provide?
Gmail was used as the email server host because its mail isn't recognised as SPAM in the mailbox unlike SendGrid.

#### What are the relevant pivotal tracker stories?
#161340174


#### Screenshots (if appropriate)
![image](https://user-images.githubusercontent.com/40101281/47862760-efd9d280-de06-11e8-96bb-aec2c3e3fdbd.png)
![image](https://user-images.githubusercontent.com/40101281/47862782-f8320d80-de06-11e8-816f-a78178767cc4.png)


#### Questions:
